invenio_banners/services/service.py

```diff
@@ -47,3 +80,34 @@ def create(self, identity, data, raise_errors=True):
         return self.result_item(
             self, identity, banner, links_tpl=self.links_item_tpl, errors=errors
         )
+
+    def delete(self, identity, id):
+        """Delete a banner from database."""
+        self.require_permission(identity, "delete")
+
+        # check if banner exists
+        banner = self.record_cls.get(id)
```

Comment:
The code duplication to retrieve a banner and trigger an exception can probably be improved by simply extracting it in a method:
```suggestion
        def _read(self, identity, id):
        """Retrieve a banner."""
        banner = self.record_cls.get(id)
        if banner is None:
            raise BannerNotExistsError(id)
        return banner
```

and then:

```
def read(self, identity, id):
        """Retrieve a banner."""
        self.require_permission(identity, "read")
        self._read(identity, id)
        return ...
```